### PR TITLE
Add ts-node dependency

### PR DESCRIPTION
**Pull request by pglushchenko-ledger**

### CODE REVIEW

**Review by glethuillier — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 .env
+.idea
```
Comment (meriadec) [inline .gitignore:2]: prefer using [a global gitignore](http://egorsmirnov.me/2015/05/04/global-gitignore-file.html) for user-specific stuff (`.idea` has nothing to do with xpub-scan files)

File: package.json [hunk 2]
```diff
@@ -6,6 +6,7 @@
     "xpub-scan": "./lib/scan.js"
   },
   "scripts": {
+    "scan": "node --require ts-node/register ./src/scan",
```
Comment (meriadec) [inline package.json:9]: what about directly

```
"scan": "ts-node ./src/scan",
```

File: src/actions/checkBalance.ts [hunk 3]
```diff
@@ -11,6 +9,8 @@ import { DerivationMode } from "../configuration/currencies";
 import { getStats, getTransactions } from "./processTransactions";
 import { TODO_TypeThis } from "../types";
 
+const sb = require("satoshi-bitcoin");
```
Comment (meriadec) [inline src/actions/checkBalance.ts:12]: intended to **not** use the JS import syntax?

File: src/actions/deriveAddresses.ts [hunk 4]
```diff
@@ -1,11 +1,12 @@
 import * as bjs from "bitcoinjs-lib";
 import * as bip32 from "bip32";
-import * as bch from "bitcoincashjs";
 import bchaddr from "bchaddrjs";
 
 import { DerivationMode } from "../configuration/currencies";
 import { configuration } from "../configuration/settings";
 
+const bch = require("bitcoincashjs");
```
Comment (meriadec) [inline src/actions/deriveAddresses.ts:8]: same comment

File: src/actions/saveAnalysis.ts [hunk 5]
```diff
@@ -12,6 +11,8 @@ import { Address } from "../models/address";
 import { TODO_TypeThis } from "../types";
 import { currencies } from "../configuration/currencies";
 
+const sb = require("satoshi-bitcoin");
```
Comment (meriadec) [inline src/actions/saveAnalysis.ts:14]: same

File: src/comparison/diffs.ts [hunk 6]
```diff
@@ -1,8 +1,8 @@
 import chalk from "chalk";
-import sb from "satoshi-bitcoin";
-
 import { Comparison } from "../models/comparison";
 
+const sb = require("satoshi-bitcoin");
```
Comment (meriadec) [inline src/comparison/diffs.ts:4]: same

File: src/configuration/currencies.ts [hunk 7]
```diff
@@ -1,4 +1,4 @@
-import coininfo from "coininfo";
+const coininfo = require("coininfo");
```
Comment (meriadec) [inline src/configuration/currencies.ts:1]: same

File: src/input/importOperations.ts [hunk 8]
```diff
@@ -1,10 +1,10 @@
 import fs from "fs";
 import chalk from "chalk";
-import sb from "satoshi-bitcoin";
-
 import { Operation } from "../models/operation";
 import { configuration } from "../configuration/settings";
 
+const sb = require("satoshi-bitcoin");
```
Comment (meriadec) [inline src/input/importOperations.ts:6]: same

File: src/modules.d.ts [hunk 9]
```diff
@@ -1,8 +0,0 @@
-declare module "satoshi-bitcoin" {
-  export function toSatoshi(n: number): number;
-  export function toBitcoin(n: number): number;
-}
-
-declare module "coininfo";
-
-declare module "bitcoincashjs";
```
Comment (meriadec) [inline src/modules.d.ts:8]: why deleting this?

**Review by meriadec — CHANGES_REQUESTED** [review]
several feedbacks
also mode was changed to 755 (before: 644) on `./src/scan.ts`, but it's not used anywhere (it's launched with `node` in package.json) => I guess this can be reverted

File: .gitignore (hunk 1 above)
Comment (pglushchenko-ledger) [reply .gitignore:2]: That is a good point. I will set up it locally

**Comment by meriadec** [comment]
just checked and in fact, the only thing you have to do is to use `--files` flag with `ts-node` (reference [here](https://github.com/TypeStrong/ts-node/issues/782#issuecomment-838013613))

so:

```
ts-node --files ./src/scan.ts
```

**Comment by pglushchenko-ledger** [comment]
> ts-node --files ./src/scan.ts

Ok. I'll try it

**Pull request closed without merging**